### PR TITLE
fix: ensure apps build consistently in docker and locally

### PR DESCRIPTION
**Pull request by ygrishajev**
This PR ensures that all the core dependencies like `react` and `next` are aligned across all monorepo apps/packages. Additionally, it updates the Docker files accordingly. With this PR, all builds and types get consistency across environments and build the same way in both Docker and locally within the context of the monorepo.

**NOTE**: This is 1 of 3 PR which should be merged together. It should be reviewed 1st and merged last after https://github.com/akash-network/cloudmos/pull/232

### CODE REVIEW

File: apps/deploy-web/src/utils/localStorage.ts [hunk 1]
```diff
@@ -1,36 +1,26 @@
...
+  const currentVersion: string = publicRuntimeConfig.version;
+  const version = getVersion();
+  const hasPreviousVersion = version && neq(currentVersion, version);
 
-  // Only apply migrations if there was a previous version
-  if (latestUpdatedVersion && neq(currentVersion, latestUpdatedVersion)) {
+  if (hasPreviousVersion) {
     Object.keys(migrations).forEach(version => {
-      if (gt(version, latestUpdatedVersion)) {
+      if (gt(version, version)) {
```
Comment (baktun14) [inline apps/deploy-web/src/utils/localStorage.ts]: maybe not the best to both have the same name :)

File: apps/landing/package.json [hunk 2]
```diff
@@ -18,30 +18,30 @@
...
-    "next-seo": "^5.5.0",
+    "next-pwa": "^5.6.0",
+    "next-seo": "^6.1.0",
     "nextjs-google-analytics": "^2.0.0",
-    "notistack": "^2.0.5",
+    "notistack": "2.0.8",
     "nprogress": "^0.2.0",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
+    "react": "^18.3.1",
```
Comment (baktun14) [inline apps/landing/package.json]: I think we should have the version without `^` so we don't have surprise updates.

Also, there was a new version that kind of introduced a weird warning about `fetchPriority` props on a DOM element. [People are saying that in `18.2.0` is fine](https://github.com/vercel/next.js/issues/65161)

File: apps/deploy-web/src/utils/localStorage.ts (hunk 1 above)
Comment (ygrishajev) [reply apps/deploy-web/src/utils/localStorage.ts]: right

File: apps/landing/package.json (hunk 2 above)
Comment (ygrishajev) [reply apps/landing/package.json]: ok, I'll try that. versions are locked anyway. but if it'll fix that warning it makes sense

Comment (ygrishajev) [reply apps/landing/package.json]: done

File: apps/deploy-web/src/utils/localStorage.ts (hunk 1 above)
Comment (ygrishajev) [reply apps/deploy-web/src/utils/localStorage.ts]: done

**Comment by ygrishajev** [comment]
I'll rebase the other 2 PRs once we're done with this one. I can also go the opposite way and merge this one and rebase the rest against main

**Review by baktun14 — APPROVED** [review]

**Review by Redm4x — APPROVED** [review]

**Pull request merged**